components/client/src/main/java/com/hotels/styx/client/connectionpool/SimpleConnectionPool.java

```diff
@@ -156,19 +162,25 @@ private void attemptBorrowConnection(MonoSink<Connection> sink, Connection conne
     }
 
     public boolean returnConnection(Connection connection) {
-        borrowedCount.decrementAndGet();
-        if (connection.isConnected()) {
-            queueNewConnection(connection);
+        if (active) {
+            borrowedCount.decrementAndGet();
+            if (connection.isConnected()) {
+                queueNewConnection(connection);
+            }
+        } else {
+            connection.close();
         }
         return false;
     }
 
     public boolean closeConnection(Connection connection) {
         connection.close();
-        borrowedCount.decrementAndGet();
-        closedConnections.incrementAndGet();
+        if (active) {
+            borrowedCount.decrementAndGet();
+            closedConnections.incrementAndGet();
 
-        newConnection();
+            newConnection();
+        }
```

Comment:
I think we should decrement the metrics regardless, and trigger a `newConnection()` only if the pool is `active`. Like so:

```
         connection.close();
         borrowedCount.decrementAndGet();
         closedConnections.incrementAndGet();
         if (active) {
             newConnection();
         }
```

The metrics might still be made available even for the closed pool (perhaps as a separate enhancement. Therefore we should keep them consistent with the reality.